app/services/tag_archiver.rb

```diff
@@ -9,13 +9,15 @@ def initialize(tag, successor)
   end
 
   def archive
-    raise "Can't archive parent tags" if tag.can_have_children?
+    raise "Can't archive Level 1 Mainstream browse page" if tag.level_one? && tag.is_a?(MainstreamBrowsePage)
+
+    raise "Can't archive Level 1 Specialist topic with published or draft subtopics" if tag.has_active_children?
```

Comment:
...in which case, this could become one single guard clause again.

```suggestion
    raise "Can't archive this tag" unless tag.can_be_archived?
```